edx/edx-enterprise
# [ENT-299] Gracefully handle pipeline timeout

**Description:** If user takes too long to complete data sharing consent workflow, direct them back to login page instead of erroring out with a 404.

Also, follow up on #86 by adding `confirmation_alert_prompt_warning` to context of account-level consent view.

**JIRA:** [ENT-299](https://openedx.atlassian.net/browse/ENT-299)

**Dependencies:** None

**Merge deadline:** April 24th, 2017

**Installation instructions:**

1. Add a file `lms/envs/private.py` with the following contents:

    ```python
    SESSION_COOKIE_AGE = 30
    ```
2. Set up a TestShib identity provider
3. Ensure that at least one TestShib username is not linked in your platform.
4. Create an EnterpriseCustomer with TestShib attached as its identity provider.
5. Ensure that the EnterpriseCustomer has data sharing consent enabled and required at login.

**Testing instructions:**

1. Register a new user using TestShib. When prompted for consent, do not provide consent, but instead close the window. After registration, the browser session is set to session-length, so it will not time out, causing the error we've seen.
2. Ensure that your browser is entirely closed.
3. Open your browser again, and enter the login flow; select TestShib, and log in with the user you already used to attempt to register.
4. When you get to the consent page, do not submit consent. Instead, wait at least 60 seconds.
5. After waiting 60 seconds, submit the consent form. Observe that you do not encounter a 404, but are instead redirected to the login screen. After the session ages out, we do not have sufficient information to point the user towards a specific SSO provider.

**Notes:**

After merging #86, we encountered an issue with account-wide consent due to a variable that was not included for the account-level consent view. This pull request adds that variable.

**Merge checklist:**

- [x] ~~Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).~~ N/A (PR does not introduce new requirements)
- [x] ~~Documentation updated (not only docstrings)~~ N/A (PR does not introduce new features)

**Post merge:**
